### ISSUE TITLE
COMMAND Health checks impossible on busybox/scratch docker containers

### ISSUE THREAD

**Issue by LinkMJB** [mesosphere/marathon#4150]
COMMAND health checks are not possible to perform on busybox/scratch docker containers when the only content that exists in the container is the service providing statically-compiled binary. The particular docker container I am testing this with is the 'swarm' docker container being run in agent mode. In agent mode no TCP ports or HTTP services are exposed so HTTP/TCP health checks are not an alternative option.
I would also like to request a generic 'heartbeat/ping' health-check feature in cases where there is no other way to determine the health of the container.
In the circumstance that the container becomes unavailable or the hosting node is rebooted, Marathon is unable to determine that the container is no longer available/healthy.
This request was initially discussed with @karlkfi via https://dcos-community.slack.com/archives/general/p1469124050002174

**Comment by gkleiman** [comment]
@LinkMJB how would that `heartbeat/ping` check work if there's no way of contacting the container?

Mesos sends Marathon a `TASK_FAILED`/`TASK_FINISHED` update when the container exits. It will also notify Marathon if the agent is disconnected from the Master (sending a `TASK_LOST` update). Marathon will then wait a configurable amount of time for the task to come back before starting a new task.

**Comment by LinkMJB** [comment]
Hey @gkleiman, I think the best way to support this would be to remove the hardcoded/implied 'sh -c' prefix to the COMMAND health check, to support statically compiled binary busybox images.

**Comment by meichstedt** [comment]
Note: This issue has been migrated to https://jira.mesosphere.com/browse/MARATHON-1794. For more information see https://groups.google.com/forum/#!topic/marathon-framework/khtvf-ifnp8.

**Comment by meichstedt** [comment]
Note: This issue has been migrated to https://jira.mesosphere.com/browse/MARATHON-1794. For more information see https://groups.google.com/forum/#!topic/marathon-framework/khtvf-ifnp8.
